sessionctx/stmtctx/stmtctx.go

```diff
@@ -306,6 +306,10 @@ type StatementContext struct {
 	// RCCheckTS indicates the current read-consistency read select statement will use `RCCheckTS` path.
 	RCCheckTS bool
 
+	// See the detail in function `NeedDisableWarmupInOptimizer`.
+	// If DisableWarmupInOptimizer is true, Optimizer doesn't call txnManger.AdviseWarmup() to warmup.
+	DisableWarmupInOptimizer bool
```

Comment:
I am a little bit curious why this is not just something belonging to `PessimisticRCTxnContextProvider`. Then `AdviseWarmup` of `PessimisticRCTxnContextProvider` can also use it to decide whether to get ts from pd.

Reply:
We can't judge whether the statement is suitable for using the last ts before physical plan is made and I mark the `DisableWarmupInOptimizer` roughly before entering optimizer for some statements which  maybe also make a tso request at `AdviseWarmup` of `PessimisticRCTxnContextProvider`.  `PessimisticRCTxnContextProvider`  judge whether makes a constant tso in `AdviseOptimizeWithPlan` by physical plan and it doesn't need a flag.